### ISSUE TITLE
haskellPackages.bencoding fails to build

### ISSUE THREAD

**Issue by Zer0-** [NixOS/nixpkgs#42540]
## Issue description
`nix-shell -p haskellPackages.bencoding` fails in the following way on NixOS 18.03:

```
these derivations will be built:
  /nix/store/rx4x3v885x7dgwywi8snl8wcdbg7mm5k-bencoding-0.4.3.0.drv
building '/nix/store/rx4x3v885x7dgwywi8snl8wcdbg7mm5k-bencoding-0.4.3.0.drv'...
setupCompilerEnvironmentPhase
Build with /nix/store/3v8p0pvcplr3bfvrnh4fyqrqk2ddyl6f-ghc-8.2.2.
ignoring (possibly broken) abi-depends field for packages
unpacking sources
unpacking source archive /nix/store/r183lb3cw8rvdg1ipa282yn9a4p3q4wc-bencoding-0.4.3.0.tar.gz
source root is bencoding-0.4.3.0
setting SOURCE_DATE_EPOCH to timestamp 1387462000 of file bencoding-0.4.3.0/tests/properties.hs
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/tmp/nix-build-bencoding-0.4.3.0.drv-0/package.conf.d -j4 -threaded
[1 of 1] Compiling Main             ( Setup.hs, /tmp/nix-build-bencoding-0.4.3.0.drv-0/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/5fy8fzqlvdnibqladihxsxi6y9b909sv-bencoding-0.4.3.0 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --docdir=/nix/store/wl17whzjcbalxa7az3qy5zlr5y1kwl9h-bencoding-0.4.3.0-doc/share/doc --with-gcc=gcc --package-db=/tmp/nix-build-bencoding-0.4.3.0.drv-0/package.conf.d --ghc-option=-optl=-Wl,-rpath=/nix/store/5fy8fzqlvdnibqladihxsxi6y9b909sv-bencoding-0.4.3.0/lib/ghc-8.2.2/bencoding-0.4.3.0 --ghc-option=-j4 --disable-split-objs --disable-library-profiling --disable-profiling --enable-shared --disable-coverage --enable-library-vanilla --enable-executable-dynamic --enable-tests --ghc-option=-split-sections --extra-lib-dirs=/nix/store/069g827lh3hrhp4vkcq3rsh5jh65pm3l-ncurses-6.0-20171125/lib --extra-lib-dirs=/nix/store/8qfd8gx0j3yzamkrbrfz5kc00h4cqd1q-gmp-6.1.2/lib --extra-lib-dirs=/nix/store/069g827lh3hrhp4vkcq3rsh5jh65pm3l-ncurses-6.0-20171125/lib
Configuring bencoding-0.4.3.0...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:948:20 in Cabal-2.0.1.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:470:12 in Cabal-2.0.1.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:570:20 in Cabal-2.0.1.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:67:5 in Cabal-2.0.1.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:174:19 in Cabal-2.0.1.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:119:27 in Cabal-2.0.1.0:Distribution.Simple
  defaultMain, called at Setup.hs:2:8 in main:Main
Setup: Encountered missing dependencies:
deepseq ==1.3.*
builder for '/nix/store/rx4x3v885x7dgwywi8snl8wcdbg7mm5k-bencoding-0.4.3.0.drv' failed with exit code 1
error: build of '/nix/store/rx4x3v885x7dgwywi8snl8wcdbg7mm5k-bencoding-0.4.3.0.drv' failed
```

### Steps to reproduce

Run `nix-shell -p haskellPackages.bencoding ` in your terminal

## Technical details

I need bencoding as a dependency for a program I am writing, as a workaround in my project's shell.nix
I had overriden this package like so:

```
  _haskellPackages = if compiler == "default"
                       then pkgs.haskellPackages
                       else pkgs.haskell.packages.${compiler};

  haskellPackages = _haskellPackages.override {
    overrides = self: super: {
      bencoding = self.callPackage ../bencoding/default.nix {};
    };
  };
```

With the working derivation for the bencoding package written like this:

```
{ mkDerivation, attoparsec, base, bytestring, containers, deepseq
, ghc-prim, hspec, integer-gmp, mtl, pretty, QuickCheck, stdenv
, text
}:
mkDerivation {
  pname = "bencoding";
  version = "0.4.3.0";
  src = ./.;
  libraryHaskellDepends = [
    attoparsec base bytestring deepseq ghc-prim integer-gmp mtl pretty
    text
  ];
  testHaskellDepends = [
    attoparsec base bytestring containers ghc-prim hspec QuickCheck
  ];
  homepage = "https://github.com/cobit/bencoding";
  description = "A library for encoding and decoding of BEncode data";
  license = stdenv.lib.licenses.bsd3;
}
```

(please note that even though the src attribute points to a cloned repository on my local machine, I have not made any modifications to the source tree, src can be changed to the upstream source of bencoding).

Now I have no idea why it doesn't work since the one defined in nixpkgs is very similar, but hopefully this is enough information for someone to fix this.

Additionally I see that bencoding is used as a dependency for some other haskellPackages, does this mean they are broken too?

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.16.12-rt5, NixOS, 18.03.132748.68e02f8ff21 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"hydra_ruma, musnix, nixos-18.03.132748.68e02f8ff21"`
 - channels(phil): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

**Comment by mpickering** [comment]
Will be fixed if https://github.com/pxqr/bencoding/pull/7 is ever resolved.

**Comment by Zer0-** [comment]
@mpickering ah you're quite right:

```
$ git status
HEAD detached at origin/fixes-for-ghc-8.0.1
```
